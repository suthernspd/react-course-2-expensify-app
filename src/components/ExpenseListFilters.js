import React from 'react';
import { connect } from 'react-redux';
import { DateRangePicker } from 'react-dates';
import { setTextFilter, setCaseSensitive, sortByDate, sortByAmount, setStartDate, setEndDate } from '../actions/filters';

export class ExpenseListFilters extends React.Component {
    state = {
        calendarFocused: null
    };
    onDatesChange = ({ startDate, endDate }) => {
        this.props.setStartDate(startDate);
        this.props.setEndDate(endDate);
    };
    onFocusChange = (calendarFocused) => {
        this.setState(() => ({ calendarFocused }));
    };
    onTextChange = (e) => {
        this.props.setTextFilter(e.target.value);
    };
    onCheckboxChange = (e) => {
        this.props.setCaseSensitive(e.target.checked);
    }
    onSortChange = (e) => {
        if (e.target.value === 'dateNew') {
            this.props.sortByDate(false);
        } else if (e.target.value === 'dateOld') {
            this.props.sortByDate(true);
        } else if (e.target.value === 'amountMax') {
            this.props.sortByAmount(false);
        } else if (e.target.value === 'amountMin') {
            this.props.sortByAmount(true);
        }
    };
    render() {
        return (
            <div className="content-container">
                <div className="input-group">
                    <div className="input-group__item">
                        <input type="text"
                        className="text-input"
                        placeholder="Search Expenses"
                        value={this.props.filters.text}
                        onChange={this.onTextChange}/>
                    </div>
                    <div className="input-group__item">
                        <div className="checkbutton">
                            <label className="checkbutton__label">Case Sensitive
                                <input type="checkbox"
                                className="checkbutton__box"
                                name="case_sensitive"
                                id="case_sensitive"
                                checked={this.props.filters.textCaseSensitive}
                                onChange={this.onCheckboxChange} />
                            </label>
                        </div>
                    </div>
                    <div className="input-group__item">
                        <select
                        className="select"
                        value={this.props.filters.sortBy}
                        onChange={this.onSortChange} >
                            <option value="dateNew">Date - New to Old</option>
                            <option value="dateOld">Date - Old to New</option>
                            <option value="amountMax">Amount - Max to Min</option>
                            <option value="amountMin">Amount - Min to Max</option>
                        </select>
                    </div>
                    <div className="input-group__item">
                        <DateRangePicker
                        startDate={this.props.filters.startDate}
                        endDate={this.props.filters.endDate}
                        onDatesChange={this.onDatesChange}
                        focusedInput={this.state.calendarFocused}
                        onFocusChange={this.onFocusChange}
                        showClearDates={true}
                        numberOfMonths={1}
                        isOutsideRange={() => false}
                        />
                    </div>
                </div>
            </div>
            );
    }
}

const mapStateToProps = (state) => ({
    filters: state.filters
});

const mapDispatchToProps = (dispatch) => ({
    setTextFilter: (text) => dispatch(setTextFilter(text)),
    setCaseSensitive: (checked) => dispatch(setCaseSensitive(checked)),
    sortByDate: (rev) => dispatch(sortByDate(rev)),
    sortByAmount: (rev) => dispatch(sortByAmount(rev)),
    setStartDate: (startDate) => dispatch(setStartDate(startDate)),
    setEndDate: (endDate) => dispatch(setEndDate(endDate))
});

export default connect(mapStateToProps, mapDispatchToProps)(ExpenseListFilters);