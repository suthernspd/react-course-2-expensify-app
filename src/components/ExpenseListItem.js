import React from 'react';
import { Link } from 'react-router-dom';
import { timestampToDate, formatCurrency } from '../playground/numFormatting';

export const ExpenseListItem = ({ id, description, amount, createdAt }) => (
    <div>
        <Link to={`/edit/${id}`}>
            <h3>{description}</h3>
        </Link>
        <p>{formatCurrency(amount,'$')} - {timestampToDate(createdAt)}</p>
    </div>
);

export default ExpenseListItem;