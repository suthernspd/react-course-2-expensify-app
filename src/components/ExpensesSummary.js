import React from 'react';
import { connect } from 'react-redux';
import { formatCurrency } from '../playground/numFormatting';
import selectExpenses from '../selectors/expenses';
import selectExpensesTotal from '../selectors/expenses-total';

export const ExpensesSummary = ({ expenseCount, expensesTotal }) => {
    return (
        <div>
        <h1>Viewing {expenseCount} expense{expenseCount === 1 ? '' : 's'} totalling {formatCurrency(expensesTotal,'$')}</h1>
        </div>
    );
};

const mapStateToProps = (state) => {
    const visibleExpenses = selectExpenses(state.expenses, state.filters)
    return {
        expenseCount: visibleExpenses.length,
        expensesTotal: selectExpensesTotal(visibleExpenses)
    };
};

export default connect(mapStateToProps)(ExpensesSummary);