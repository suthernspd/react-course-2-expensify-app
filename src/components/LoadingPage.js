import React from 'react';


const LoadingPage = () => (
    <div className="loader">
        <p>Loading...</p>
        <img className="loader__image" src="/images/loader.gif" />
    </div>
)

export default LoadingPage;