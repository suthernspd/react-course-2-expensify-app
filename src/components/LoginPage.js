import React from 'react';
import { connect } from 'react-redux';
import { startLogin } from '../actions/auth';

export const LoginPage = ({ startLogin }) => (
    <div className="box-layout">
        <div className="box-layout__box">
            <h1 className="box-layout__title">Expensify</h1>
            <p>It's time to get your expenses under control.</p>
            <button className="button" onClick={startLogin}>Login with Google</button>
        </div>
    </div>
);

const mapDispatchToProps = (dispatch) => ({
    startLogin: () => dispatch(startLogin())
});

export default connect(undefined, mapDispatchToProps)(LoginPage);



/* 
export default class LoginPage extends React.Component {
    constructor(props) {
        super(props);

        this.state = {
            userName: props.user ? props.user.userName : '',
            password: props.user ? props.user.password : '',
            error: ''
        };
    }
    onUserNameChange = (e) => {
        const userName = e.target.value;
        this.setState(() => ({ userName }));
    };
    onPasswordChange = (e) => {
        const password = e.target.value;
        this.setState(() => ({ password }));
    };
    onSubmit = (e) => {
        e.preventDefault();

        if (!this.state.userName || !this.state.password) {
            this.setState(() => ({ error: 'Please provide a valid username and password.' }));
        } else {
            this.setState(() => ({ error: '' }));
            this.props.onSubmit({
                userName: this.state.userName,
                password: this.state.password
            });
        }
    };

    render() {
        return (
            <div>
                {this.state.error && <p>{this.state.error}</p>}
                <form onSubmit={this.onSubmit}>
                    <input
                        type="text"
                        placeholder="User Name"
                        autoFocus
                        value={this.state.userName}
                        onChange={this.onUserNameChange}
                    />
                    <input
                        type="password"
                        placeholder="Password"
                        value={this.state.password}
                        onChange={this.onPasswordChange}
                    />
                    <button onClick={startLogin}>Login</button>
                </form>
            </div>
        )
    }
} */