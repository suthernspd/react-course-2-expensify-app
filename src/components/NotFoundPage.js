import React from 'react';
import { Link } from 'react-router-dom';

const NotFoundPage = () => (
    <div>
        404 no findy - <Link to="/dashboard">go back</Link>
    </div>
);

export default NotFoundPage;