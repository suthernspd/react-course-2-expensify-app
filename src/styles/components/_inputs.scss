.text-input {
    border: 1px solid #cacccd;
    height: 50px;
    padding: $s-size;
    font-size: $font-size-medium;
}

.select {
    @extend .text-input;
    font-weight: 400;
    line-height: 1.15;
    font-family: Helvetica, Arial, sans-serif;
}

.checkbutton {
    @extend .text-input;
}

.checkbutton__label {
    display: block;
    padding-left: 15px;
    text-indent: -15px;
}

.checkbutton__box {
    @extend .text-input;
    width: 26px;
    height: 13px;
    padding: 0;
    margin: 0;
    vertical-align: bottom;
    position: relative;
    top: -5px;
    *overflow: hidden;
}

.textarea {
    @extend .text-input;
    height: 10rem;
}