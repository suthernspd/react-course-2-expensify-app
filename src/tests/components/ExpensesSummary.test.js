import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpensesSummary';

test('should correctly render ExpensesSummary with 1 expense', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={1} expensesTotal={235}/>);
    expect(wrapper).toMatchSnapshot();
});

test('should correctly render ExpensesSummary with many expenses', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={22} expensesTotal={235123456}/>);
    expect(wrapper).toMatchSnapshot();
});