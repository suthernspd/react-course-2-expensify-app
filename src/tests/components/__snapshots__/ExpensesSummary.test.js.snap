// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render ExpensesSummary with 1 expense 1`] = `
<div>
  <h1>
    Viewing 
    1
     expense
     totalling 
    $2.35
  </h1>
</div>
`;

exports[`should correctly render ExpensesSummary with many expenses 1`] = `
<div>
  <h1>
    Viewing 
    22
     expense
    s
     totalling 
    $2,351,234.56
  </h1>
</div>
`;
